docs/clients.md

```diff
@@ -1,10 +1,42 @@
 # Clients
 
-There are multiple ways to communicate with a Traffic Generator via the OTG API:
- <!-- TODO add links from bold items to paragraphs in Clients section -->
-* **otgen** command-line tool is an easy way to start with
-* **snappi** library to accelerate development of *Test Programs* written in Python or Go
-* **direct REST or gRPC calls** as an alternative to using *snappi*
-* **custom** OTG client applications
-
-[snappi](https://pypi.org/project/snappi/) and [gosnappi](https://pkg.go.dev/github.com/open-traffic-generator/snappi/gosnappi) provide client side API libraries for the OTG specifications for Python and Golang respectively.  For other languages, SDKs can be built using [OpenAPI Generator](https://github.com/OpenAPITools/openapi-generator) (for REST API) or [protobuf tools](https://github.com/protocolbuffers/protobuf) (for gRPC).  
\ No newline at end of file
+## Overview
+ 
+To successfully use an OTG-based Traffic Generator, you need to be able to execute the following tasks over the OTG API:
+ 
+* Prepare a **Configuration** and apply it to a Traffic Generator
+* **Control** states of the configured objects like Protocols or Traffic Flows
+* Collect and analyze **Metrics** reported by the Traffic Generator
+ 
+It is a job of an **OTG Client** to perform these tasks by communicating with a Traffic Generator via the OTG API. There are different types of such clients, and the choice between them depends on how and where you want to use a Traffic Generator.
+ 
+## Command-line Utilities
+ 
+Command-line utilities is the fastest way to execute an OTG API request, especially in the environment without access to Python or Go development toolchains. They are also the easiest option to use by someone who is just starting with the OTG.
+ 
+The most basic utility for any kind of REST API calls, including OTG, is **`curl`**. But, it leaves all the knowledge of OTG to the user. See the [curl](clients/curl.md) section for examples of using it with OTG.
+ 
+On the other side of the spectrum is **`otgen`**. This command-line utility comes as part of OTG toolkit. It is capable of manipulating a wide range of OTG features while hiding a lot of complexity from a user. `otgen` is an excelent choice for a first time OTG user to become familiar with this ecosystem. It is also a great choice for an advanced user who needs to execute a common OTG scenario in an environment without a development toolchain. Continue reading about this utility in the [otgen](clients/otgen.md) section.
```

Comment:
Question : On the other side of the spectrum OR On the other end of the spectrum?